packages/twenty-front/src/modules/ui/layout/modal/components/__stories__/ConfirmationModal.stories.tsx

```diff
@@ -29,5 +60,117 @@ export const InputConfirmation: Story = {
     confirmationPlaceholder: 'email@test.dev',
     ...Default.args,
   },
-  decorators: Default.decorators,
+};
+
+export const CloseOnEscape: Story = {
+  args: {
+    modalId: 'confirmation-modal',
+    title: 'Escape Key Test',
+    subtitle: 'This modal should close when pressing the Escape key.',
+    confirmButtonText: 'Confirm',
+    onClose: closeMock,
+  },
+  play: async ({ canvasElement }) => {
+    const canvas = within(canvasElement);
+
+    await canvas.findByText('Escape Key Test');
+
+    closeMock.mockClear();
```

Comment:
I think you don't need to clear the mocks. The testing framework is responsible for clearing mocks between each test. I've tested it, and it appears to be working.